### DISCUSSION TITLE
DriverBase detecting motors in a twisted way (SPIKE PRIME, ADB)

### DISCUSSION THREAD

**Discussion by afarago** [pybricks/support#318] [Q&A] [answered]
Hi,
Sorry for the dummy question - I managed to install pybricks on SPIKE PRIME via https://beta.pybricks.com/.
Adding code via https://code.pybricks.com/ works fine, connection via BT is perfect.

When I instantiate a DriveBase for Advanced Driving Base core model straight startes to turn, while turn goes straight - as if either A or E motor would be in opposite direction.

```python
left_motor = Motor(Port.A) 
right_motor = Motor(Port.E) 
robot = DriveBase(left_motor, right_motor, wheel_diameter=88, axle_track=145)
robot.straight(100)
#robot.turn(90)
```

Any hints for me?

**Comment by dlech** [comment, marked as answer]
Hi Attila! You will need to change the direction of one of the motors so that the positive direction of each motor matches how the motors are mounted on your drive base.

```python
# for example
left_motor = Motor(Port.A, positive_direction=Direction.COUNTERCLOCKWISE) 
```

See [docs](https://docs.pybricks.com/en/latest/pupdevices/motor.html) for the default positive motor rotations.

**Reply by afarago** [reply]
Hi David!
Perfect - for some reason I was looking in the DriveBase docu for some control parameter, but have not assumed the solution to be found around the individual motor initialization. Many thanks for your valuable help.

For the record - this is the setting for the advanced driving base that worked for me

```python
hub = PrimeHub()
left_motor = Motor(Port.A, positive_direction=Direction.COUNTERCLOCKWISE) 
right_motor = Motor(Port.E) 
robot = DriveBase(left_motor, right_motor, wheel_diameter=88, axle_track=144)
```

ADB model: https://education.lego.com/en-us/lessons/prime-competition-ready/assembling-an-advanced-driving-base

**Reply by laurensvalk** [reply]
Glad to hear it worked out @afarago!

How do you like Pybricks so far?

**Reply by afarago** [reply]
Love it and will coach three girls for WRO open to dive into their project with pybricks in the upcoming 2 months.

Started downloading to build locally - bit struggling, will retry on the weekend as I see big potential in this awesome project.

**Reply by laurensvalk** [reply]
That's awesome - looking forward to hear about your experience using it.
